### ISSUE TITLE
Contributors & Acknolwedgements 

### ISSUE THREAD

**Issue by BalzaniEdoardo** [flatironinstitute/ccn-template#3]
Discussion points:

**Acknowledging contributors:**
* How do we acknowledge contributors? (simply list names or specify in what way they contributed, e.g. documentation, bugfix, resolve issue...)
* What does constitute enough contribution to be listed? (e.g. typos no, docstrings maybe, closing issue for sure)
* Criteria should be clearly stated as well as guidelines for how to contribute see [sklearn](https://scikit-learn.org/stable/developers/contributing.html) for examples

**Referencing literature and authors**
* How do we acknowledge the literature we are basing our repository upon (e.g. J Pillow for the GLM library)
* Should we reach out to scientists that developed a method that will be included in our code-base? how?

**Comment by EricThomson** [comment]
Good point: also, mentioning how to cite the software (when applicable).

**Comment by billbrod** [comment]
I think we should aim to give users something to cite for each package. In addition to DOIs through [zenodo](https://zenodo.org/) (which will mint one for each release, so users can specify the version), we can submit a software paper. I'm inclined to do it through [pyopensci](https://www.pyopensci.org/) or [JOSS](https://joss.theoj.org/) (a pyopensci review leads to a JOSS paper, but not every package is within scope for pyopensci, so some should be submitted directly to JOSS), rather than a standard journal because pyopensci/JOSS will actually review the code and focus on the code, docs, tests, etc. I'll be submitting to pyopensci for plenoptic soon, so can update you on what that process is like.

Then, "how to cite" includes the software paper, the specific version, and the specific papers describing each method. I'm going to put something for this together for pyrtools and plenoptic soon.

Additionally, we could consider doing something like nipreps / fmriprep: automatically create a natural language description of the workflow (see [here](https://www.nipreps.org/intro/transparency/#citation-boilerplates)). This would be a template string where bits get plugged in based on which steps are run, and the appropriate parameter values are inserted. This might only be relevant for Caiman, which is the most pipeline-like of our packages (I don't think it would be hard to add this to mesmerize, which is already tracking the parameters people are using), but I like the idea of doing this as a way of making it easy for users to cite everything and increase understandability of the workflow.

**Comment by EricThomson** [comment]
Dropping this here so I don't lose track of it:
https://intersect-training.org/collaborative-git/04-git-workflows/index.html

**Comment by billbrod** [comment]
For contributors, [all contributors](https://allcontributors.org/) has a way of recognizing contributors who haven't contributed code (and so don't show up on github's automatically generated list)

**Comment by billbrod** [comment]
https://usrse.slack.com/archives/C01HRSZ4VNJ/p1690924075237679
